src/RageSoundReader_Chain.cpp

```diff
@@ -186,19 +186,19 @@ void RageSoundReader_Chain::Finish()
 	m_iActualSampleRate = GetSampleRateInternal();
 	if( m_iActualSampleRate == -1 )
 	{
-		for (RageSoundReader *it : m_apLoadedSounds)
+		for (auto &pSound : m_apLoadedSounds)
 		{
-			RageSoundReader_Resample_Good *pResample = new RageSoundReader_Resample_Good( it, m_iPreferredSampleRate );
-			it = pResample;
+			RageSoundReader_Resample_Good *pResample = new RageSoundReader_Resample_Good( pSound, m_iPreferredSampleRate );
+			pSound = pResample;
 		}
 
 		m_iActualSampleRate = m_iPreferredSampleRate;
 	}
 
 	/* Attempt to preload all sounds. */
-	for (RageSoundReader *it : m_apLoadedSounds)
+	for (auto &pSound : m_apLoadedSounds)

```

Comment:
I wonder if this one is actually necessary. I don't *think* we're modifying pSound here (unless `PreloadSound` does stuff). Try reverting and testing to see if it's necessary.

Reply:
okie

Reply:
yeahhh this one is still needed